RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClusterizerConditionsESProducerAlpaka.cc

```diff
@@ -0,0 +1,278 @@
+                                                                                       std::vector<uint32_t>& detID,
+                                                                                       std::vector<uint16_t>& iPair,
+                                                                                       std::vector<uint16_t>& noise,
+                                                                                       std::vector<float>& gain) {
+    // Lambda to lookup channel index and set the invThickness array
+    auto setInvThickness = [&](uint16_t fed, uint8_t channel, float val) {
+      invthick[channelIndex(fed, channel)] = val;
+    };
+
+    // connected: map<DetID, std::vector<int>>
+    // map of KEY=detid DATA=vector of apvs, maximum 6 APVs per detector module :
+    const auto& connected = quality.cabling()->connected();
+    // detCabling: map<DetID, std::vector<const FedChannelConnection*>
+    // map of KEY=detid DATA=vector<FedChannelConnection>
+    const auto& detCabling = quality.cabling()->getDetCabling();
+
+    for (const auto& conn : connected) {
+      const auto det = conn.first;
+      if (!quality.IsModuleBad(det)) {
+        const auto detConn_it = detCabling.find(det);
+
+        if (detCabling.end() != detConn_it) {
+          for (const auto& chan : (*detConn_it).second) {
+            if (chan && chan->fedId() && chan->isConnected()) {
+              const auto chan_detID = chan->detId();
+              const auto chan_fedID = chan->fedId();
+              const auto chan_fedCh = chan->fedCh();
+              const auto chan_apvPairNumber = chan->apvPairNumber();
+
+              // Fill the data structures
+              // Note: the channelIndex is used to access the data structures
+              detID[channelIndex(chan_fedID, chan_fedCh)] = chan_detID;
+              iPair[channelIndex(chan_fedID, chan_fedCh)] = chan_apvPairNumber;
+              setInvThickness(chan_fedID, chan_fedCh, siStripClusterTools::sensorThicknessInverse(chan_detID));
+
+              auto offset = STRIPS_PER_FEDCH * chan_apvPairNumber;
+
+              for (auto strip = 0; strip < STRIPS_PER_FEDCH; ++strip) {
+                const auto gainRange = gains->getRange(det);
+
+                const auto detstrip = strip + offset;
+                const uint16_t strip_noise = SiStripNoises::getRawNoise(detstrip, noises.getRange(det));
+                const auto strip_gain = SiStripGain::getStripGain(detstrip, gainRange);
+                const auto bad = quality.IsStripBad(quality.getRange(det), detstrip);
+
+                // gain is actually stored per-APV, not per-strip
+                // setStrip_(chan_fedID, chan_fedCh, detstrip, noise, gain, bad);
+                gain[apvIndex(chan_fedID, chan_fedCh, detstrip)] = strip_gain;
+                if (bad) [[unlikely]] {
+                  noise[stripIndex(chan_fedID, chan_fedCh, detstrip)] |= badBit;
```

Comment:
Is it intended to keep the original value of `noise[...]` and set only the `badBit` ?

If
```suggestion
                  noise[stripIndex(chan_fedID, chan_fedCh, detstrip)] = badBit;
```
is valid, it would be simpler and more efficient.